components/scream/src/share/tests/atm_process_tests.cpp

```diff
@@ -429,4 +606,117 @@ TEST_CASE ("subcycling") {
   }
 }
 
+TEST_CASE ("diagnostics") {
+
+  //TODO: This test needs a field manager so that changes in Field A are seen everywhere.
+  using namespace scream;
+
+  // A world comm
+  ekat::Comm comm(MPI_COMM_WORLD);
+
+  // A time stamp
+  util::TimeStamp t0 ({2022,1,1},{0,0,0});
+
+  // Create a grids manager
+  auto gm = create_gm(comm);
+
+  // Use the AddOne process to generate the field data to use with diagnostic
+  ekat::ParameterList params_proc_A;
+  params_proc_A.set<std::string>("Process Name", "AddOne");
+  params_proc_A.set<std::string>("Grid Name", "Point Grid");
+  auto ap1     = std::make_shared<AddOne>(comm,params_proc_A);
+  ap1->set_grids(gm);
+
+  // Use the AddTwo process to generate the field data to use with diagnostic
+  ekat::ParameterList params_proc_B;
+  params_proc_B.set<std::string>("Process Name", "AddTwo");
+  params_proc_B.set<std::string>("Grid Name", "Point Grid");
+  auto ap2     = std::make_shared<AddTwo>(comm,params_proc_B);
+  ap2->set_grids(gm);
+
+  // Create the identity diagnostic
+  ekat::ParameterList params_identity;
+  params_identity.set<std::string>("Diagnostic Name", "DiagIdentity");
+  params_identity.set<std::string>("Grid Name", "Point Grid");
+  auto diag_identity = std::make_shared<DiagIdentity>(comm,params_identity);
+  diag_identity->set_grids(gm);
+
+  // Create the sum diagnostic
+  ekat::ParameterList params_sum;
+  params_sum.set<std::string>("Diagnostic Name", "DiagSum");
+  params_sum.set<std::string>("Grid Name", "Point Grid");
+  auto diag_sum = std::make_shared<DiagSum>(comm,params_sum);
+  diag_sum->set_grids(gm);
+
+  // Create the fail diagnostic
+  ekat::ParameterList params_fail;
+  params_fail.set<std::string>("Diagnostic Name", "DiagFail");
+  params_fail.set<std::string>("Grid Name", "Point Grid");
+  auto diag_fail = std::make_shared<DiagFail>(comm,params_fail);
+  diag_fail->set_grids(gm);
+
+  // Create fields for ap1
+  for(const auto& req : ap1->get_required_field_requests()) {
+    Field f(req.fid);
+    f.allocate_view();
+    f.deep_copy(0);
+    f.get_header().get_tracking().update_time_stamp(t0);
+    // Set the required and computed fields for the process and for the diagnostic(s)
+    ap1->set_required_field(f.get_const());
```

Comment:
This seems like an overkill. Can't you just do `f.deep_copy(2);`, and do away with the AddTwo process?

Reply:
I think you might have missed this comment.